wsidicom/series/pyramids.py

```diff
@@ -109,6 +110,36 @@ def _group_instances_into_pyramids(
                 )
             else:
                 ext_depth_of_field = None
+
+            def _cs_matches(cs_a, cs_b):
+                return (
+                    (cs_a[0] - cs_b[0]) ** 2 + (cs_a[1] - cs_b[1]) ** 2
+                ) <= settings.pyramids_origin_threshold**2 and cs_a[2] == cs_b[2]
```

Comment:
Maybe move this to a method in the ImageCoordinateSystem-class that takes the threshold?